gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ManagedController.java

```diff
@@ -128,6 +133,29 @@ DeliveryConfig deleteManifest(@PathVariable("name") String name) {
     return keelService.deleteManifest(name);
   }
 
+  @ApiOperation(value = "Validate a delivery config manifest", response = Map.class)
+  @PostMapping(
+      path = "/delivery-configs/validate",
+      consumes = {APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE},
+      produces = {APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE})
+  ResponseEntity<Map> validateManifest(@RequestBody DeliveryConfig manifest) {
+    try {
+      return ResponseEntity.ok(keelService.validateManifest(manifest));
+    } catch (RetrofitError e) {
+      if (e.getResponse().getStatus() == 400) {
+        try {
+          return ResponseEntity.badRequest()
+              .body(objectMapper.readValue(e.getResponse().getBody().in(), Map.class));
+        } catch (IOException ioe) {
```

Comment:
I'm wondering if `IOException` is the only exception we can get here? Does `objectMapper.readValue` return a different exception? (or maybe even just NPE?)

Reply:
Nothing that it declares explicitly, which means if there are any it would be unchecked exceptions. I could expand this to catch `Exception` without any harm, I suppose?

Reply:
I think it might be safer?

Reply:
Changed! 👍